Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -646,558 +587,373 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        /// <summary>
-        /// Tries to identify the string and return an object of that class.
-        /// </summary>
-        /// <param name="addr">String to parse.</param>
-        /// <param name="result">IPObject to return.</param>
-        /// <returns><c>true</c> if the value parsed successfully, <c>false</c> otherwise.</returns>
-        private static bool TryParse(string addr, out IPObject result)
+        /// <inheritdoc/>
+        public bool TryParseInterface(string intf, [NotNullWhen(true)] out IReadOnlyList<IPData>? result)
         {
-            if (!string.IsNullOrEmpty(addr))
+            var resultList = new List<IPData>();
+            if (string.IsNullOrEmpty(intf) || _interfaces is null)
             {
-                // Is it an IP address
-                if (IPNetAddress.TryParse(addr, out IPNetAddress nw))
-                {
-                    result = nw;
-                    return true;
-                }
+                result = resultList.AsReadOnly();
+                return false;
+            }
 
-                if (IPHost.TryParse(addr, out IPHost h))
+            // Match all interfaces starting with names starting with token
+            var matchedInterfaces = _interfaces.Where(s => s.Name.Equals(intf, StringComparison.OrdinalIgnoreCase)).OrderBy(x => x.Index).ToList();
+            if (matchedInterfaces.Count > 0)
+            {
+                _logger.LogInformation("Interface {Token} used in settings. Using its interface addresses.", intf);
```

Comment:
Is this log crucial? Otherwise we could just use a single LINQ expression here and reduce LOC. E.g.
```C#
result = _interfaces
    .Where(s => s.Name.Equals(intf, StringComparison.OrdinalIgnoreCase)
        && ((IsIPv4Enabled && iface.Address.AddressFamily == AddressFamily.InterNetwork)
            || (IsIPv6Enabled && iface.Address.AddressFamily == AddressFamily.InterNetworkV6)))
    .OrderBy(x => x.Index)
    .ToArray();
return result.Length > 0;

Reply:
I suppose you could check the length and then log accordingly